src/sagemaker/workflow/steps.py

```diff
@@ -287,6 +287,16 @@ def __init__(
             )
             warnings.warn(msg)
 
+        self.job_name = None
+        if estimator.source_dir or estimator.entry_point:
+            # By default, `Estimator` will upload the local code to an S3 path
+            # containing a timestamp. This causes cache misses whenever a
+            # pipeline is updated, even if the underlying script hasn't changed.
+            # To avoid this, hash the contents of the training script and include it
+            # in the `job_name` passed to the `Estimator`, which will be used
+            # instead of the timestamped path.
+            self.job_name = self._generate_code_upload_path()
```

Comment:
but `job_name` later will be excluded from the definition file right? how the backend decide when to use cache or not tho?